Mathlib/AlgebraicGeometry/EllipticCurve/Affine/Point.lean

```diff
@@ -487,6 +487,43 @@ lemma zero_def : 0 = (.zero : W'.Point) :=
 lemma some_ne_zero {x y : R} (h : W'.Nonsingular x y) : Point.some h ≠ 0 := by
   rintro (_ | _)
 
+/-- The `X` coordinate of a given point. For the point at infinity, this returns `0`
+(junk value). -/
+def x : W'.Point → R
+  | 0 => 0
+  | @some _ _ _ x _ _ => x
+
+/-- The `Y` coordinate of a given point. For the point at infinity, this returns `0`
+(junk value). -/
+def y : W'.Point → R
+  | 0 => 0
+  | @some _ _ _ _ y _ => y
+
+/-- Custom recursor for `[NeZero P]`. -/
+@[elab_as_elim]
+def neZeroRec {C : W'.Point → Sort*} (h₁ : {x y : R} → (h : W'.Nonsingular x y) → C (some h)) :
+    (P : W'.Point) → [NeZero P] → C P
+  | 0, h₀ => (h₀.ne _ rfl).elim
+  | some h, _ => h₁ h
+
+lemma equation_x_y (P : W'.Point) [NeZero P] : W'.Equation P.x P.y :=
+  P.neZeroRec fun h ↦ h.1
+
+lemma equation_x_y' (P : W'.Point) [NeZero P] : P.y ^ 2 + W'.a₁ * P.x * P.y + W'.a₃ * P.y
+    = P.x ^ 3 + W'.a₂ * P.x ^ 2 + W'.a₄ * P.x + W'.a₆ :=
+  (equation_iff ..).1 P.equation_x_y
+
+/-- The partial derivative `∂W/∂X` of the Weierstrass cubic at a given point `P`. -/
+def px (P : W'.Point) : R :=
+  W'.a₁ * P.y - (3 * P.x ^ 2 + 2 * W'.a₂ * P.x + W'.a₄)
+
+/-- The partial derivative `∂W/∂Y` of the Weierstrass cubic at a given point `P`. -/
+def py (P : W'.Point) : R :=
+  2 * P.y + W'.a₁ * P.x + W'.a₃
```

Comment:
I think I might have overlooked this last time, but why do you need these instead of using `polynomialX` and `polynomialY`? They could have just been `polynomialX.evalEval P.x P.y` and `polynomialY.evalEval P.x P.y`. Unless it's a computability issue, in which case I'm happy to change `polynomialX` and `polynomialY`.

Reply:
I think last time I said that I don't really like indirectly going through eval, and I forgot what your reply was.

Reply:
I'm saying you should remove this definition.

Reply:
@kckennylau If you want to keep these definitions you should probably either define them in terms of `polynomialX/Y` as @Multramate suggests, or if you want to keep the current form, keeping them computable, then you should provide lemmas relating them with `polynomialX/Y` 